### ISSUE TITLE
Should emit error about private struct before unresolved name

### ISSUE THREAD

**Issue by jhasse** [rust-lang/rust#26839]
I was trying to use std::thread::sleep and created the following code:

```rust
use std::time::duration::Duration;
use std::thread::sleep;

fn main() {
    sleep(Duration(1));
}
```

The (only) error message is:

```
<anon>:5:11: 5:19 error: unresolved name `Duration`
<anon>:5     sleep(Duration(1));
                   ^~~~~~~~
error: aborting due to previous error
```

But it would have helped me a lot more if it would be:

```
<anon>:1:5: 1:34 error: struct `Duration` is private
<anon>:1 use std::time::duration::Duration;
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
error: aborting due to previous error
```

**Comment by mitsuhiko** [comment]
The same problem also happens with trying to use a public struct like a function.

**Comment by leodasvacas** [comment]
1.14 output:
```
error: module `duration` is private
 --> <anon>:1:5
  |
1 | use std::time::duration::Duration;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
This is bug is fixed.
